src/services/laodeai/laodeai.js

```diff
@@ -0,0 +1,101 @@
+import cheerio from 'cheerio';
+import got from 'got';
+import { getCommandArgs } from '../../utils/command.js';
+import { cleanURL, fetchDDG } from '../../utils/http.js';
+import { generateImage } from '../snap/utils.js';
+import { stackoverflow } from './stackoverflow.js';
+import { gist } from './gist.js';
+import { wikipedia } from './wikipedia.js';
+
+// list of handlers, also used to filter valid sites
+const VALID_SOURCES = {
+  'stackoverflow.com': stackoverflow,
+  'gist.github.com': gist,
+  'wikipedia.com': wikipedia,
+};
+
+/**
+ * @param {import('telegraf').Telegraf} context
+ * @returns {Promise<void>}
+ */
+async function laodeai(context) {
+  const query = getCommandArgs('laodeai', context);
+  if (!query) return;
+
+  const { body: ddgBody, statusCode: ddgStatusCode } = await fetchDDG(got, query);
+
+  if (ddgStatusCode !== 200) {
+    await context.reply('Error getting search result.');
+    return;
+  }
+
+  const $ = cheerio.load(ddgBody);
+  const validSources = $('.web-result')
+    .map((_, el) => {
+      const $$ = cheerio.load($.html(el));
```

Comment:
kenapa harus diload lagi? kenapa nggak di for each aja? @elianiva

Reply:
sek, `el` ini instancenya cheerio?

Reply:
iya, element instance. kamu ga dapet typingsnya?

Reply:
https://cheerio.js.org/classes/Cheerio.html#map cek telegram